common/src/colors.cpp

```diff
@@ -557,27 +557,27 @@ static const unsigned char VIRIDIS_LUT[] =
   247, 230, 31 ,
   249, 231, 33 ,
   251, 231, 35 ,
-  254, 231, 36 ,
+  254, 231, 36
 };
 
 /// Number of colors in Glasbey lookup table
-static const std::size_t GLASBEY_LUT_SIZE = sizeof (GLASBEY_LUT) / (sizeof (GLASBEY_LUT[0]) * 3);
+constexpr std::size_t GLASBEY_LUT_SIZE = GLASBEY_LUT.size() / 3;
 
 /// Number of colors in Viridis lookup table
-static const std::size_t VIRIDIS_LUT_SIZE = sizeof (VIRIDIS_LUT) / (sizeof (VIRIDIS_LUT[0]) * 3);
+constexpr std::size_t VIRIDIS_LUT_SIZE = VIRIDIS_LUT.size() / 3;
 
-static const unsigned char* LUTS[] = { GLASBEY_LUT, VIRIDIS_LUT };
-static const std::size_t LUT_SIZES[] = { GLASBEY_LUT_SIZE, VIRIDIS_LUT_SIZE };
+constexpr std::array<const std::array<unsigned char, 256 * 3>*, 2> LUTS = { &GLASBEY_LUT, &VIRIDIS_LUT };
+constexpr std::array<std::size_t, 2> LUT_SIZES = { GLASBEY_LUT_SIZE, VIRIDIS_LUT_SIZE };
 
 template<typename pcl::ColorLUTName T> pcl::RGB
 pcl::ColorLUT<T>::at (std::size_t color_id)
 {
   assert (color_id < LUT_SIZES[T]);
   pcl::RGB color;
-  color.r = LUTS[T][color_id * 3 + 0];
-  color.g = LUTS[T][color_id * 3 + 1];
-  color.b = LUTS[T][color_id * 3 + 2];
-  return (color);
+  color.r = LUTS[T]->at(color_id * 3 + 0);
```

Comment:
This will cause exception handling code to be generated. Let's not do that
```cpp
(*LUTS[T])[idx]  // looks ugly, but less machine code is generated
```

Reply:
I'm thinking about to let it be and add `operator[]` to synchronize the API with `std::array`. What is your opinion about this?

Reply:
This makes sense

Reply:
Okay, this is not possible:
```
error: overloaded 'operator[]' cannot be a static member function
      static RGB operator[] (std::size_t color_id);
                 ^
```